mermithid/processors/misc/MultiGasComplexLineShape.py

```diff
@@ -0,0 +1,3748 @@
+        amplitude_fit = params[1]
+        sigma_fit = params[2]
+        scatter_proportion_fit = list(params[3:2+N]) + [1- sum(params[3:2+N])]
+        total_counts_fit = amplitude_fit
+
+        perr = m_binned.np_errors()
+        B_field_fit_err = perr[0]
+        amplitude_fit_err = perr[1]
+        sigma_fit_err = perr[2]
+        scatter_proportion_fit_err = list(perr[3:2+N]) + [np.sqrt(sum(perr[3:2+N]**2))]
+        total_counts_fit_err = amplitude_fit_err
+
+        fit_Hz = self.spectrum_func_composite_gaussian_lorentzian_fixed_survival_probability(bins_Hz, eff_array, *params)
+        fit_keV = ComplexLineShapeUtilities.flip_array(fit_Hz)
+        bins_keV = ConversionFunctions.Energy(bins_Hz, B_field_fit)/1000
+        bins_keV = ComplexLineShapeUtilities.flip_array(bins_keV)
+        reduced_chi2 = m_binned.fval/(len(fit_Hz)-m_binned.nfit)
+
+        if print_params == True:
+            output_string = '\n'
+            output_string += 'Reduced chi^2 = {:.2e}\n'.format(reduced_chi2)
+            output_string += '-----------------\n'
+            output_string += 'B field = {:.8e}'.format(B_field_fit)+' +/- '+ '{:.4e} T\n'.format(B_field_fit_err)
+            output_string += '-----------------\n'
+            output_string += 'Amplitude = {}'.format(round(amplitude_fit,2))+' +/- {}'.format(round(amplitude_fit_err,2)) + '\n'
+            output_string += '-----------------\n'
+            output_string += 'sigma = {:.2e}'.format(sigma_fit) + ' +/- {:.4e}\n'.format(sigma_fit_err)
+            output_string += '-----------------\n'
+            for i in range(len(self.gases)):
+                output_string += '{} Scatter proportion \n= '.format(self.gases[i]) + "{:.6e}".format(scatter_proportion_fit[i])\
+                +' +/- ' + "{:.2e}".format(scatter_proportion_fit_err[i])+'\n'
+                output_string += '-----------------\n'
+        elapsed = time.time() - t
+        output_string += 'Fit completed in '+str(round(elapsed,2))+'s'+'\n'
+        dictionary_of_fit_results = {
+        'output_string': output_string,
+        'perr': perr,
+        'bins_keV': bins_keV,
+        'fit_keV': fit_keV,
+        'bins_Hz': bins_Hz,
+        'fit_Hz': fit_Hz,
+        'B_field_fit': B_field_fit,
+        'B_field_fit_err': B_field_fit_err,
+        'sigma_fit': sigma_fit,
+        'sigma_fit_err': sigma_fit_err,
+        'amplitude_fit': amplitude_fit,
+        'amplitude_fit_err': amplitude_fit_err,
+        'data_hist_freq': data_hist_freq,
+        'reduced_chi2': reduced_chi2
+        }
+        return dictionary_of_fit_results
+
+    #fitting with composite gaussian lorentzian resolution function and the scatter fractions fixed for some gas species
+    def make_spectrum_composite_gaussian_lorentzian_fixed_survival_probability_partially_fixed_scatter_proportion(self, scatter_proportion, sigma, emitted_peak='shake'):
```

Comment:
how is scatter proportion partially fixed?

Reply:
The fixed scatter proportions are not fit parameters. Only the free scatter proportions are in the fit parameters. This is most convenient to see at around the line 2150. A commented is added there.